src/quicklist.c

```diff
@@ -2033,6 +2023,21 @@ int quicklistTest(int argc, char *argv[], int accurate) {
             quicklistRelease(ql);
         }
 
+
+        TEST("NEXT plain node")
+        {
+            packed_threshold = 1;
+            quicklist *ql = quicklistNew(-2, options[_i]);
+            quicklistPushHead(ql, "hello1", 6);
+            quicklistPushHead(ql, "hello2", 6);
+            quicklistPushHead(ql, "hello3", 6);
+            quicklistEntry entry;
+            quicklistIter *iter = quicklistGetIterator(ql, AL_START_TAIL);
+            quicklistNext(iter, &entry);
```

Comment:
i think testing just one call to `Next` in this test is a miss.
let's do a full iteration to cover the entire list (and even verify the last `Next` returns 0.
maybe it would also be a good idea to mix plain and non plain nodes?
i.e. set the threshold to 3, and add:
```
hello1
hello2
h3
h4
hello5
```

Reply:
@oranagra  what kind of test in corrupt-dump.tcl  do you want ? I see that in the existing test "corrupt payload: fuzzer findings - valgrind negative malloc" I get to the new if   :         
if (data == NULL || (encoded_len == 0)) {
                if (encoded_len == 0)
                    zfree(data);
                decrRefCount(o);
                return NULL;
            }

Reply:
@oranagra means to add `restore list 0 "\x12\x01\x01\x00\n\x00\x8f\xc6\xc0W\x1c\n\xb3<" replace` to the corrupt-dump.tcl test.
This test will not fail until you add `(encoded_len == 0)`, after that it will return `*Bad data format*` error.

Reply:
which test in corrupt-dump.tcl should I add it to ?

Reply:
Add a test like this:
```tcl
test {corrupt payload: quicklist encoded_len is 0} {
    start_server [list overrides [list loglevel verbose use-exit-on-panic yes crash-memcheck-enabled no] ] {
        catch { r restore _list 0 "\x12\x01\x01\x00\n\x00\x8f\xc6\xc0W\x1c\n\xb3<" replace } err
        assert_match "*Bad data format*" $err
        r ping
    }
}
```